### PR TITLE
Noetic test dev

### PR DESCRIPTION
**Pull request by Axelito95**
This is a fix to import leg_detector in noetic version. opencv -> opencv2

### CODE REVIEW

**Comment by DLu** [comment]
Thanks for working on this. 

Right now the CI isn't working because [bfl](https://wiki.ros.org/noetic/Migration#Use_Orocos_KDL_and_BFL_rosdep_keys) is still in the package.xml .

**Comment by Axelito95** [comment]
> 
> 
> Thanks for working on this.
> 
> Right now the CI isn't working because [bfl](https://wiki.ros.org/noetic/Migration#Use_Orocos_KDL_and_BFL_rosdep_keys) is still in the package.xml .

i don't know what is CI, i saw that in your package you already added bfl migration key [bfl](https://wiki.ros.org/noetic/Migration#Use_Orocos_KDL_and_BFL_rosdep_keys) . i had bfl library problem but i fixed this on my pc installed the library manually : sudo apt install liborocos-bfl-dev

up : i saw that you right in package.xml is bfl depedency o.o, i will fix this soon

**Comment by DLu** [comment]
My apologies, CI stands for continuous integration which is the system which ensures the code will compile for everyone.

**Comment by Axelito95** [comment]
> My apologies, CI stands for continuous integration which is the system which ensures the code will compile for everyone.

Np ! just my bad that i don't know CI acronyme.Added new bfl key dependency

i see that the build doesn't pass because : 
Checking for module 'orocos-bfl'
--   No package 'orocos-bfl' found

On my pc i fixed it with : sudo apt-get install -y liborocos-bfl-dev

File: leg_detector/src/calc_leg_features.cpp [hunk 1]
```diff
@@ -34,8 +34,10 @@
 
 #include <leg_detector/calc_leg_features.h>
 
-#include <opencv/cxcore.h>
-#include <opencv/cv.h>
+//#include <opencv/cxcore.h>
+//#include <opencv/cv.h>
```
Comment (DLu) [inline leg_detector/src/calc_leg_features.cpp]:
```suggestion
```

File: leg_detector/src/leg_detector.cpp [hunk 2]
```diff
@@ -37,9 +37,11 @@
 #include <leg_detector/laser_processor.h>
 #include <leg_detector/calc_leg_features.h>
 
-#include <opencv/cxcore.h>
-#include <opencv/cv.h>
-#include <opencv/ml.h>
+//#include <opencv/cxcore.hpp>
+//#include <opencv/cv.h>
+//#include <opencv/ml.h>
```
Comment (DLu) [inline leg_detector/src/leg_detector.cpp]:
```suggestion
```

File: leg_detector/src/leg_detector.cpp [hunk 3]
```diff
@@ -96,12 +99,13 @@ class SavedFeature
 
   double reliability, p;
 
-  Stamped<Point> position_;
+  tf::Stamped<tf::Point> position_;
+  //Stamped<Point> position_;
```
Comment (DLu) [inline leg_detector/src/leg_detector.cpp]:
```suggestion
```

File: leg_detector/src/train_leg_detector.cpp [hunk 4]
```diff
@@ -35,9 +35,11 @@
 #include <leg_detector/laser_processor.h>
 #include <leg_detector/calc_leg_features.h>
 
-#include <opencv/cxcore.h>
-#include <opencv/cv.h>
-#include <opencv/ml.h>
+//#include <opencv/cxcore.hpp>
+//#include <opencv/cv.h>
+//#include <opencv/ml.h>
```
Comment (DLu) [inline leg_detector/src/train_leg_detector.cpp]:
```suggestion
```

**Review by DLu — COMMENTED** [review]
@Axelito95 Do you mind if I remove these commented lines?

File: leg_detector/src/leg_detector.cpp (hunk 2 above)
Comment (Axelito95) [reply leg_detector/src/leg_detector.cpp]: yes sure, you can! I let it just to show to others what libraries can be replaced but I saw that they can see the changes from commits

**Comment by DLu** [comment]
Thank you very much @Axelito95 for figuring this out!

**Comment by Axelito95** [comment]
> 
> 
> Thank you very much @Axelito95 for figuring this out!

You are welcome ! And thank you too for your working ! Happy new year ;) !

**Pull request merged**